src/cpp/session/SessionContentUrls.cpp

```diff
@@ -95,13 +95,13 @@ Error contentFileInfo(const std::string& contentUrl,
    if (newline != std::string::npos)
       *pTitle = pTitle->substr(0, newline);
    
-   // get file parameter
+   // get file parameter (if any)
    std::string contentFile = http::util::fieldValue(fields, "file");
    if (contentFile.empty())
-       return systemError(ENOENT, ERROR_LOCATION);
-   *pFilePath = contentUrlPath().complete(contentFile);
+      return Success();
```

Comment:
This does seem like an error condition since the out parameter `*pFilePath` is not returned. Is there a reason we're invoking it without a file in the first place?

Reply:
AFAIK this is the default behavior for all GWT classes inheriting from UrlContentEditingTarget:

https://github.com/rstudio/rstudio/blob/9c3e611e856c36a317664218f9cbdf67a428510b/src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/urlcontent/UrlContentEditingTarget.java#L346-L357

But this fails for the Data Viewer since there isn't an associated file to be removed with this API (?)

Do you think it would be better to just override `onDismiss()` in `DataEditingTarget` and make it a no-op?

Reply:
In fact that's what we're already doing in the object explorer so I'll make that change.

Reply:
This is the 'shape' of a content URL for the data viewer:

```
"grid_resource/gridviewer.html?env=package%3Adatasets&obj=mtcars&cache_key=6FF93F22"
```

Since it doesn't have a `file` parameter, attempts to remove it will fail.